libs/detectors/detector.py

```diff
@@ -29,7 +29,7 @@ def __init__(self, config):
             self.detector = DummyDetector(self.config)
         elif self.device in ["x86", "x86-gpu"]:
             from libs.detectors.x86.detector import Detector as X86Detector
-            self.detector = X86Detector(self.config)
+            self.detector = X86Detector(self.config, source)
```

Comment:
I think that we need to send the source to all the constructors, not only the X86

Reply:
Of course, this commit was just a proof of concept in order to validate these changes. I will generalize it.